Update documentation to cover installation of all dependencies (#961)
jupyterlab/jupyter-ai: docs/source/users/index.md

pip install jupyter-ai
```

> In some cases, subsidiary packages to be installed, e.g., required by some model providers and some file formats. To ensure such cases are handled, to install every dependency to get access to all models currently supported by `jupyter-ai`, you should run (and then reload Jupyter Lab):

Reviewer: ```suggestion
In some cases, subsidiary packages need to be installed, e.g., because they are required by some model providers and some file formats.
To ensure such cases are handled you can install every dependency which will give you access to all models currently supported by `jupyter-ai`
To install every dependency, run (and then restart JupyterLab):
```

The sentences appear incomplete. I am not sure how to make it sound best, please do adjust the suggestions as you seem it fit best.

Reviewer: Also "subsidiary packages" seem very technical, not sure if "third-party" would be better here?

Author: @krassowski - Thanks for the helpful comments! I have implemented these.

> In some cases, third-party packages need to be installed, for example, because they are required by some model providers and some file formats. To ensure such cases are handled you can install every dependency which will give you access to all models currently supported by `jupyter-ai`. To install every dependency, run (and then restart JupyterLab):

Reviewer 2: ```suggestion
You may need to install third-party packages, for example, to use some model providers and some file formats with Jupyter AI. To handle all supported use cases, you can install every dependency, which will give you access to all models currently supported by `jupyter-ai`. To install every dependency, run the following command, and then restart JupyterLab:
```
